xbmc/FileItem.cpp

```diff
@@ -124,7 +124,7 @@ CFileItem::CFileItem(const CVideoInfoTag& movie)
   SetFromVideoInfoTag(movie);
 }
 
-void CFileItem::FillMusicInfoTag(const CPVRChannelPtr channel, const CPVREpgInfoTagPtr tag)
+void CFileItem::FillMusicInfoTag(const CPVRChannelPtr& channel, const CPVREpgInfoTagPtr& tag)
```

Comment:
passing a smart pointer by ref does not make much sense to me. pass the object it holds by ref or have the shared pointer increment by passing by value. passng a smart pointer by ref can hit you badly in a multithreaded environment

Reply:
for mutable refs, sure. for const refs, you do not get any additional safety by doing an increase/decrease cycle. it's not being stored anywhere, we just read data and return.

Reply:
> for mutable refs, sure. for const refs, you do not get any additional safety by doing an increase/decrease cycle.

 If you want to ensure that the object behind the shared pointer is kept alive while the method is called, passing it by value is eactly what you want. As @FernetMenta said, this is generally a good approach in multithreaded environment.

Reply:
please. read the surrounding code. read what i wrote. i know very well how shared pointers work. if shit manage to remove it from under our feet in this helper function that reads and immediately returns, something else is wrong. you are just stressing the ref counting for absolutely no reason here. but whatever, i will just remove this commit and let you guys act on generics and not logic.

Reply:
Got it. See my EDIT...

Reply:
Please readd the commit.

Reply:
round and round it goes. readded

Reply:
Thanks. You were just to fast for me rethinking what shit I wrote in the first place. ;-)